src/cpu/test.rs

```diff
@@ -814,6 +831,79 @@ fn test_shift() {
     assert_eq!(cpu.get_flag(Flag::H), false);
     assert_eq!(cpu.get_flag(Flag::C), true);
 
+    cpu.set_reg(Register8Bit::B, 0x95);
+    cpu.set_flag(Flag::C, true);
+    cpu.step();
+    assert_eq!(cpu.get_reg(Register8Bit::B), 0x2b);
+    assert_eq!(cpu.get_flag(Flag::Z), false);
+    assert_eq!(cpu.get_flag(Flag::N), false);
+    assert_eq!(cpu.get_flag(Flag::H), false);
+    assert_eq!(cpu.get_flag(Flag::C), true);
+
+    cpu.set_reg(Register8Bit::B, 0x85);
+    cpu.set_flag(Flag::C, true);
+    cpu.step();
+    assert_eq!(cpu.get_reg(Register8Bit::B), 0x0b);
+    assert_eq!(cpu.get_flag(Flag::Z), false);
+    assert_eq!(cpu.get_flag(Flag::N), false);
+    assert_eq!(cpu.get_flag(Flag::H), false);
+    assert_eq!(cpu.get_flag(Flag::C), true);
+
+    cpu.set_reg(Register8Bit::B, 0x81);
```

Comment:
Let's use the test cases from the manual for all of these to be safe.

Reply:
These are your tests for rla,rlca,rra,rrca copies lol. There are no tests lined out for other rotations.

Reply:
check again my duuuuuuude